### PR TITLE
airbyte-ci: ClickPipelineContext handles dagger logging

### PR DESCRIPTION
**Pull request by alafanechere**
## What
`ClickPipelineContext` users rely on the `get_dagger_client` method to get the dagger client as a singleton.
We originally exposed a `log_output` parameter to this method, but as it returns a singleton dagger client, the passed `log_output` is not taken into account if the singleton already exists.

## How
The `get_dagger_client` defines the log output path according to the `show_dagger_logs` params.
If `show_dagger_logs` is true: show the logs in the stdout, otherwise log to a file.

Next:
* Remove the `show_dagger_logs` param/options and always log to a file
* Users can see the dagger logs if they're using a `dagger run` wrapped command
* Broaden the use of `ClickPipelineContext` to have a consistent logging management implementation

### CODE REVIEW

**Comment by alafanechere** [comment]
Current dependencies on/for this PR:
* `master`
  * **PR #33419** <a href="https://app.graphite.dev/github/pr/airbytehq/airbyte/33419?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/airbytehq/airbyte/33419?utm_source=stack-comment).

File: airbyte-ci/connectors/pipelines/pipelines/models/contexts/click_pipeline_context.py [hunk 1]
```diff
@@ -76,15 +79,18 @@ def __init__(self, **data: dict[str, Any]):
...
         """
         Get (or initialize) the Dagger Client instance.
         """
         if not self._dagger_client:
             async with self._dagger_client_lock:
                 if not self._dagger_client:
+                    if self.params.get("show_dagger_logs", False):
+                        log_output = sys.stdout
+                    else:
+                        log_output, self._click_context().obj["dagger_logs_path"] = self._create_dagger_client_log_file()
```
Comment (bnchrch) [inline airbyte-ci/connectors/pipelines/pipelines/models/contexts/click_pipeline_context.py:92]: Nice change! 

One house keeping thing

**1. Move this logic to its own method**
```py
def get_log_output(self):
    if self.params.get("show_dagger_logs", False):
        log_output = sys.stdout
    else:
        log_output, self._click_context().obj["dagger_logs_path"] = self._create_dagger_client_log_file()

    return log_output
```

**Review by bnchrch — COMMENTED** [review]
Good change! Just have a question on where this should live.

File: airbyte-ci/connectors/pipelines/pipelines/models/contexts/click_pipeline_context.py [hunk 2]
```diff
@@ -97,6 +103,14 @@ async def get_dagger_client(self, pipeline_name: Optional[str] = None, log_outpu
         assert self._dagger_client, "Error initializing Dagger client"
         return self._dagger_client.pipeline(pipeline_name) if pipeline_name else self._dagger_client
 
+    def _create_dagger_client_log_file(self) -> Tuple[io.FileIO, str]:
```
Comment (bnchrch) [inline airbyte-ci/connectors/pipelines/pipelines/models/contexts/click_pipeline_context.py:106]: ❓  Instead of adding this logic here, could we not hoist it into `airbyte_ci.py` or `dagger_run.py` and make it shared between all pipelines?

Comment (alafanechere) [reply airbyte-ci/connectors/pipelines/pipelines/models/contexts/click_pipeline_context.py:106]: It's something I considered but as the log file is tightly coupled to the client I prefer to keep it where the client is instantiated. 
I'm also conscious that we currently have inconsistencies in the way we handle dagger client logging in our commands.
I created this [issue](https://github.com/airbytehq/airbyte/issues/33487) to consolidate the approach. 
TLDR; all our commands should get their dagger client from `ClickPipelineContext`

Comment (bnchrch) [reply airbyte-ci/connectors/pipelines/pipelines/models/contexts/click_pipeline_context.py:112]: That sounds fine then!

At that point in time we should be in a good place to introduce the GlobalSettings pattern from aircmd,

**Review by bnchrch — APPROVED** [review]
lgtm!
